beacon-chain/blockchain/chain_info.go

```diff
@@ -187,6 +188,25 @@ func (s *Service) HeadState(ctx context.Context) (state.BeaconState, error) {
 	return s.cfg.StateGen.StateByRoot(ctx, s.headRoot())
 }
 
+// HeadStateReadOnly returns the read only head state of the chain.
+// If the head is nil from service struct,
+// it will attempt to get the head state from DB.
+func (s *Service) HeadStateReadOnly(ctx context.Context) (state.ReadOnlyBeaconState, error) {
```

Comment:
Can we add a trivial unit test to make sure that all paths are covered here and protect from our future selfs?